From 19e2e3ad5a960c493b7b2c86db0ff27fc6420d35 Mon Sep 17 00:00:00 2001
Message-Id: <19e2e3ad5a960c493b7b2c86db0ff27fc6420d35.1423872525.git.feitong.yi@intel.com>
In-Reply-To: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
References: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Sat, 27 Dec 2014 12:32:31 +0530
Subject: [PATCH 09/95] SQUASHME! [VPG]: drm/i915: Fix to avoid access to
 disabled power wells

This is a miss in Sanitize Display PHY patch, this patch adds a check to
avoid processing of disabled PHY in intel_display_power_enabled_unlocked.

The initial Sanitize Display PHY based on VBT patch was introduced in:
commit 827f5e4b780b9d7342fc22d30ec8d5f559c8019e
Author: Deepak S <deepak.s@intel.com>
Date:   Sat Nov 29 08:25:24 2014 +0530

    FOR_UPSTREAM [VPG]: drm/i915: Sanitize Display PHY based on VBT

Issue: GMIN-3619
Change-Id: I657cd7cb9084742df8055def7970f8f6f39a194d
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 drivers/gpu/drm/i915/intel_pm.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
index f097813..bf9bf66 100644
--- a/drivers/gpu/drm/i915/intel_pm.c
+++ b/drivers/gpu/drm/i915/intel_pm.c
@@ -6564,7 +6564,7 @@ bool intel_display_power_enabled_unlocked(struct drm_i915_private *dev_priv,
 	is_enabled = true;
 
 	for_each_power_well_rev(i, power_well, BIT(domain), power_domains) {
-		if (power_well->always_on)
+		if (power_well->always_on || power_well->power_domain_disabled)
 			continue;
 
 		if (!power_well->hw_enabled) {
-- 
1.7.9.5

